components/SearchBar.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react';
+import { GooglePlacesAutocomplete } from 'react-native-google-places-autocomplete';
+import { REACT_APP_GOOGLE_PLACES_API } from 'react-native-dotenv';
+import Location from '../classes/location';
+import styles from '../constants/SearchBarStyling';
+
+const apiKey = REACT_APP_GOOGLE_PLACES_API;
+type searchBarProps = {
+  setMapLocation(location: Location): void;
+};
+
+type searchBarState = {};
+
+class SearchBar extends Component<searchBarProps, searchBarState> {
+  setMapLocation = (lat, lng) => {
+    const { setMapLocation } = this.props;
+    setMapLocation(new Location(lat, lng));
+  };
+
+  render() {
+    return (
+      <GooglePlacesAutocomplete
+        placeholder="Search"
+        minLength={2} // minimum length of text to search
+        autoFocus={false}
+        listViewDisplayed="false"
+        keyboardShouldPersistTaps="always"
+        returnKeyType="search" // Can be left out for default return key https://facebook.github.io/react-native/docs/textinput.html#returnkeytype
+        keyboardAppearance="light" // Can be left out for default keyboardAppearance https://facebook.github.io/react-native/docs/textinput.html#keyboardappearance
+        fetchDetails
+        renderDescription={row => row.description} // custom description render
+        onPress={(data, details = null) => {
+          // 'details' is provided when fetchDetails = true
+          this.setMapLocation(details.geometry.location.lat, details.geometry.location.lng);
+        }}
+        getDefaultValue={() => ''}
+        query={{
+          // available options: https://developers.google.com/places/web-service/autocomplete
+          key: apiKey,
+          language: 'en', // language of the results
+          location: '45.496628, -73.578804', // Centered around hall building
+          radius: '80000',
```

Comment:
we can probably minimize the radius to make it more relevant for concordia

Reply:
ill make it a radius of 10km. Is that good? Would you rather have something smaller?

Reply:
We also nee to consider Loyola campus